src/application/application.cpp

```diff
@@ -242,7 +245,9 @@ void Application::Predict() {
     objective_fun_->Init(train_data_->metadata(), train_data_->num_data());
     boosting_->Init(&config_, train_data_.get(), objective_fun_.get(),
                     Common::ConstPtrInVectorWrapper<Metric>(train_metric_));
-    boosting_->RefitTree(pred_leaf);
+
+
+    boosting_->RefitTree(&pred_leaf.front(), nrow, pred_leaf.size() / nrow);
```

Comment:
`pred_leaf.front()` has undefined behavior if the vector is empty :eyes:

```suggestion
    boosting_->RefitTree(pred_leaf.data(), nrow, pred_leaf.size() / nrow);
```